src/spam-protection/google-recaptcha.ts

```diff
@@ -54,13 +63,25 @@ export default class GoogleRecaptcha {
         }, { isEqual: (a, b) => a === b });
     }
 
-    load(containerId: string, sitekey: string): Promise<void> {
-        return this.googleRecaptchaScriptLoader.load()
-            .then(recaptcha => {
-                this._event$ = this._memoized(recaptcha, sitekey, document.getElementById(containerId));
-
-                this._recaptcha = recaptcha;
-            });
+    async load(
+        containerId: string,
+        sitekey: string,
+        size: ReCaptchaV2.Size = 'invisible',
+        onSuccess: (token: string) => void = noop,
+        onError: (error: Error) => void = noop
+    ): Promise<void> {
+        const recaptcha = await this.googleRecaptchaScriptLoader.load();
+
+        this._event$ = this._memoized(recaptcha, sitekey, document.getElementById(containerId), size);
+        this._recaptcha = recaptcha;
+
+        this._event$.subscribe(result => {
+            if (result && result.token) {
+                onSuccess(result.token);
```

Comment:
I think there's some logic in `execute` that handles the scenario where reCaptcha puzzle is dismissed by the shopper before it is solved. That scenario won't trigger the error callbacks that the reCaptcha client provides, so we need some custom code to handle it. Do we not need to handle that scenario anymore?

Reply:
The regular (not invisible catpcha) works differently, the execute method doesn't need to be called.